### ISSUE TITLE
no-restricted-syntax false positive

### ISSUE THREAD

**Issue by vsemozhetbyt** [eslint/eslint#8733]
* **ESLint Version:** 4.0.0
* **Node Version:** 8.1.1
* **npm Version:** 5.0.3

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
This template is used for finding certain code patterns (`RegExp` with flags in this case)
```js
'use strict';

module.exports = {
  root: true,
  parserOptions: { ecmaVersion: 2017, sourceType: 'script' },
  env: { es6: true, node: true, browser: true },

  rules: {
    'no-restricted-syntax': ['error', 'Literal[regex.flags=/./]'],
  },
};
```

**What did you do? Please include the actual source code causing the issue.**
```js
'use strict';

console.log(/a/);
console.log(/a/i);
```

**What did you expect to happen?**
```console
 4:13  error  Using 'Literal[regex.flags=/./]' is not allowed  no-restricted-syntax
```

**What actually happened? Please include the actual, raw output from ESLint.**
```console
  1:1   error  Using 'Literal[regex.flags=/./]' is not allowed  no-restricted-syntax
  4:13  error  Using 'Literal[regex.flags=/./]' is not allowed  no-restricted-syntax
```

**Comment by vsemozhetbyt** [comment]
This variant works as expected:
```js
  'no-restricted-syntax': ['error', 'Literal[regex.flags="i"]'],
```

**Comment by vsemozhetbyt** [comment]
These variants gives no error messages at all:
```js
'no-restricted-syntax': ['error', 'Literal[regex.flags=/\w/]'],
'no-restricted-syntax': ['error', 'Literal[regex.flags=/[\w]/]'],
```

And this one has the same issue as the first one (false positive on non-RegExp literal):
```js
'no-restricted-syntax': ['error', 'Literal[regex.flags=/[gimuy]/]'],
```

**Comment by not-an-aardvark** [comment]
This is happening because the regex is matched against `null`, which is casted to a string. This seems like a bug in [esquery](https://github.com/estools/esquery). As a workaround, you can use

```
Literal[regex][regex.flags=/./]
```

**Comment by platinumazure** [comment]
ESQuery 1.0.1 has been released with a fix for this issue. I'll open a PR to consume it in ESLint.
